src/server/debug.rs

```diff
@@ -228,48 +228,67 @@ impl Debugger {
-            })?;
-
-        let new_conf_ver = region.get_region_epoch().get_conf_ver();
-        let old_conf_ver = old_region.get_region_epoch().get_conf_ver();
-
-        // If the store is not in peers, or it's still in but its peer_id
-        // has changed, we know the peer is marked as tombstone success.
-        let scheduled = region
-            .get_peers()
-            .iter()
-            .find(|p| p.get_store_id() == store_id)
-            .map_or(true, |p| p.get_id() != peer_id);
-
-        if new_conf_ver > old_conf_ver && scheduled {
-            let wb = WriteBatch::new();
-            // Here we can keep the other metas as original.
-            box_try!(write_peer_state(db, &wb, &old_region, PeerState::Tombstone));
+        let wb = WriteBatch::new();
+        let handle = box_try!(get_cf_handle(db.as_ref(), CF_RAFT));
+        let mut ret = Vec::with_capacity(regions.len());
+
+        {
+            let fix_one_region = |region: Region| {
+                let id = region.get_id();
+                let key = keys::region_state_key(id);
+                let store_id = self.get_store_id()?;
+
+                let mut region_state = box_try!(db.get_msg_cf::<RegionLocalState>(CF_RAFT, &key))
+                    .expect("Can't find RegionLocalState");
+
+                let peer_id = region_state
+                    .get_region()
+                    .get_peers()
+                    .iter()
+                    .find(|p| p.get_store_id() == store_id)
+                    .map(|p| p.get_id())
+                    .ok_or_else(|| {
+                        Error::Other("RegionLocalState doesn't contains the peer itself".into())
+                    })?;
+
+                let old_conf_ver = region_state.get_region().get_region_epoch().get_conf_ver();
+                let new_conf_ver = region.get_region_epoch().get_conf_ver();
+
+                // If the store is not in peers, or it's still in but its peer_id
+                // has changed, we know the peer is marked as tombstone success.
+                let scheduled = region
+                    .get_peers()
+                    .iter()
+                    .find(|p| p.get_store_id() == store_id)
+                    .map_or(true, |p| p.get_id() != peer_id);
+
+                if new_conf_ver > old_conf_ver && scheduled {
+                    let wb = WriteBatch::new();
+                    region_state.set_state(PeerState::Tombstone);
+                    box_try!(wb.put_msg_cf(handle, &key, &region_state));
+                    return Ok(());
+                }
+                Err(box_err!("The peer is still in target peers"))
+            };
+
+            for region in regions {
+                let id = region.get_id();
+                if let Err(e) = fix_one_region(region) {
+                    ret.push((id, e));
+                }
+            }
+        }
+
+        if ret.is_empty() {
```

Comment:
not empty?

Reply:
`ret.is_empty()` is true means error occurs when we update rocksdb into write batch. In this case, we shouldn't write anything into disk. So I think it's correct.

Reply:
but here your logic is if empty, then write to RocksDB.

Btw, don't use this for the error assumption, if you meet error, handle the error.